### PR TITLE
{FEATURE} - Feature update /code clean for update 1.9

### PR DESCRIPTION
**Pull request by jupster**
Note I attend to use this as a beta until current update 1.9 comes out so If you aren't prepared to merge this for a while that is okay. 
## Description

Added code from @TheSpartaPT for optional mods in https://github.com/Foxlider/FASTER/pull/95 plus added their code from https://github.com/Foxlider/FASTER/pull/96 which adds me being less scared ill hit 'Delete All' by accident, plus a few other things.

Any feel free to add comments.
## Motivation and Context

Closes https://github.com/Foxlider/FASTER/pull/140
Closes https://github.com/Foxlider/FASTER/pull/138
Closes https://github.com/Foxlider/FASTER/pull/95
Closes https://github.com/Foxlider/FASTER/pull/96
Closes https://github.com/Foxlider/FASTER/issues/24
(Already added in 1.9 but cleaned code up) https://github.com/Foxlider/FASTER/pull/129
Closes https://github.com/Foxlider/FASTER/issues/144
Closes https://github.com/Foxlider/FASTER/issues/107
## How Has This Been Tested?

This has been tested to the best of my ability. The build is being run on a couple of servers successfully at this time. All build's working as expecting, the only think i think may be off is spacing but ill likely fix it later on. Not a major priority. Only issues i had run into was with 144 not working. It seems to be working flawlessly now.
## Types of changes

- [X] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:

- [X] My code follows the code style of this project.
- [X] I have read the **CONTRIBUTING** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: FASTER/ViewModel/ModsViewModel.cs [hunk 1]
```diff
@@ -135,14 +136,26 @@ internal void DeleteMod(ArmaMod mod)
...
+        {
+            var answer = await DialogCoordinator.ShowInputAsync(this, "Are you sure you want to delete all mods?", "Write \"yes\" and press OK if you wish to continue.");
+
+            if (string.IsNullOrEmpty(answer) || !answer.Equals("yes"))
+                return;
+
+            Analytics.TrackEvent("Mods - Clicked DeleteAllMods", new Dictionary<string, string>
+            {
+                {"Name", Properties.Settings.Default.steamUserName}
+            });
```
Comment (Foxlider) [inline FASTER/ViewModel/ModsViewModel.cs:158]: Missing }

**Review by Foxlider — CHANGES_REQUESTED** [review]
Not compiling, missing `}`

**Comment by jupster** [comment]
[Updated as it was 2 commits behind](https://github.com/jupster/FASTER/releases/tag/1.9d_142_beta)

**Comment by jupster** [comment]
@Foxlider any chance we can get this reviewed so we it can be merged to Update 1.9. 

Thanks

**Comment by Foxlider** [comment]
I got a week of exams but this weekend I'll look at it

**Review by Foxlider — APPROVED** [review]
awyep

**Pull request merged**